File: week_02/src/main/java/Question20.java
/*
Author = Zaid Nayeb
Date   = 18/11/2025

 20. Count digits of a number
Description: Find how many digits are in a number.
Example: Input:
786
Output:
3

*/
public class Question20 {
    public static int countDigits(int number){
        if (number < 0){
            number *= -1;
        }
        int digits = 0;
        while (number > 0){
            number /= 10;
            digits++;
        }
        return digits;
    }
}

File: week_04/src/Question_20.js
/* 
Author  :   Zaid Nayeb
Date    :   2025/12/07

20.	Check Balanced Parentheses  
Input: "{[()]}" → Output: Balanced 
"{()[]}" → Output:  Balanced

*/
console.log(isBalanced("{()[]}"));


function isBalanced(parantheses) {
    const stack = [];
    const map = {
        ')': '(',
        '}': '{',
        ']': '['
    }

    for (let char of parantheses) {
        if (char === '(' || char === '{' || char === '[') {
            stack.push(char);
        } else {
            if (stack.pop() !== map[char]) {
                return false;
            }
        }
    }
    return stack.length === 0;
}

module.exports = isBalanced;